components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/util/APIFileUtils.java

```diff
@@ -125,8 +126,8 @@ public static void writeStringAsJsonToFile(String content, String filePath) thro
     public static void writeToFile(String path, String content) throws APIMgtDAOException {
 
         try (FileOutputStream fileOutStream = new FileOutputStream(path);
-                StringReader stringReader = new StringReader(content);
-                OutputStreamWriter writer = new OutputStreamWriter(fileOutStream, StandardCharsets.UTF_8);) {
```

Comment:
With this code block, output stream will not be flushed or closed. String content will be remained in buffer for sometime and it may consume memory. We need to close it properly after write done. Also instead of copying we can directly write.